app/scripts/modules/core/src/delivery/details/StageFailureMessage.tsx

```diff
@@ -76,21 +94,38 @@ export class StageFailureMessage extends React.Component<IStageFailureMessagePro
         );
       }
 
-      const currentState = ReactInjector.$state.current;
-      return (
-        <div className="row">
-          <div className="col-md-12">
-            <div className="alert alert-danger">
-              <div>
-                Stage <UISref to={currentState.name} params={{ step: failedStageIndex }}>
+      if (failedStage && (failedStageIndex || failedStepIndex)) {
+        const currentState = ReactInjector.$state.current;
+        const params: any = {
+          'step': failedStepIndex
+        };
+        if (failedExecutionId) {
+          params.executionId = failedExecutionId;
+        }
+        if (failedStageIndex) {
```

Comment:
don't rely on truthy here - if `failedStageIndex` is `0`, it won't get set.

Reply:
That's true.